classes/webservice/WebserviceKey.php

```diff
@@ -83,7 +83,24 @@ public static function keyExists($key)
 
     public function delete()
     {
-        return parent::delete() && ($this->deleteAssociations() !== false);
+        $result = parent::delete() && ($this->deleteAssociations() !== false);
+
+        if ($result) {
+            PrestaShopLogger::addLog(
+                Context::getContext()->getTranslator()->trans(
+                    'Webservice key deleted by employee "%d" %s %s (%s)',
+                    [
+                        Context::getContext()->employee->id,
+                        Context::getContext()->employee->firstname,
+                        Context::getContext()->employee->lastname,
+                        Context::getContext()->employee->email,
+                    ],
+                    'Admin.Advparameters.Feature'
+                )
+            );
```

Comment:
Employee identifier is the last argument of `PrestaShopLogger::addLog`
```suggestion
            PrestaShopLogger::addLog(
                Context::getContext()->getTranslator()->trans(
                    'Webservice key deleted',
                    [],
                    'Admin.Advparameters.Feature'
                ),
                1,
                0,
                'WebserviceKey',
                (int) $this->id,
                false,
                (int) Context::getContext()->employee->id
            );
```
I don't found existing constants for Severity Levels...

Reply:
Thank you I apply